app/src/main/java/com/babylon/wallet/android/presentation/onboarding/restore/mnemonics/RestoreMnemonicsScreen.kt

```diff
@@ -74,15 +74,26 @@ fun RestoreMnemonicsScreen(
         state = state,
         onBackClick = viewModel::onBackClick,
         onSkipClicked = viewModel::onSkipClick,
+        onSkipMainSeedPhraseClicked = viewModel::onSkipMainSeedPhraseClicked,
         onSubmitClick = {
-            if (!state.isShowingEntities) {
-                context.biometricAuthenticate { authenticated ->
-                    if (authenticated) {
-                        viewModel.onSubmit()
+            when (state.screenType) {
+                RestoreMnemonicsViewModel.State.ScreenType.NoMainSeedPhrase -> {
+                    context.biometricAuthenticate { authenticated ->
```

Comment:
Need to confirm with iOS if we need to generate mnemonic, otherwise biometric won't be needed here.

Reply:
Yes we authenticate here as we cannot generate mnemonic without asking for biometric